valm08-sp2

valm08 opened the pull request:
Pilotní verze - zatím pouze stahovaní PDF souborů.

https://eso.vse.cz/~valm08/sp2/

valm08 committed:
valm08-sp2

Pilotní verze - zatím pouze stahovaní PDF souborů.

https://eso.vse.cz/~valm08/sp2/

valm08 committed:
version 02

Většina funkcionality.
Zdá se, že PDF soubory nelze stahovat přes AJAX kvůli CORS (ale citace ano?), takže jsou prostě stahovány přímo z endpointu a nejsou zazipované.

https://eso.vse.cz/~valm08/sp2/

nvbach91 commented on www/valm08/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,142 @@
+  function renderFavourites() {
+    for(let i=0; i<localStorage.length; i++) {
+    let key = localStorage.key(i);
+    $(outputField).append(`${localStorage.getItem(key)} <br>`);
```
chyba aktualizace DOMu v cyklu,

dale tyhle polozky maji klikatelne, pri kliknuti se provede stahovani

nvbach91 commented on www/valm08/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,142 @@
+  renderFavourites();
+
+
+  function spaceToUnderscore (text) {
```
na tohle prece neni potreba funkce :)

nvbach91 commented on www/valm08/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,142 @@
+    let input = inputField.val();
+    input = input.trim();
+    input = spaceToUnderscore(input);
+    downloadFile(`https://en.wikipedia.org/api/rest_v1/page/pdf/${input}/a4/mobile`);
```
tenhle base URL se da ulozit do promenne

nvbach91 commented on www/valm08/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,142 @@
+  });
+
+
+  $(desktopPdfButton).click(function(){
```
proc je tady `$()`?

nvbach91 commented on www/valm08/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,142 @@
+    $.ajax({
+      method: 'GET',
+      url: `https://en.wikipedia.org/api/rest_v1/data/citation/bibtex/${input}`,
+      success: (resp) => {
```
pouzivejte spis Promise

nvbach91 commented on www/valm08/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,142 @@
+
+
+  $(favouriteButton).click(function(){
+    for(let i=0; i<localStorage.length; i++) {
```
prubezne si kod naformatujte

nvbach91 commented on www/valm08/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,142 @@
+  //based on https://stackoverflow.com/questions/1964839/how-can-i-create-a-please-wait-loading-animation-using-jquery
+  $(document).on({
+    ajaxStart: function(){
+      $(outputField).text("");
```
tohle je zbytecny

nvbach91 commented on www/valm08/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,142 @@
+        return;
+      };
+      };
+    localStorage.setItem(localStorage.length,inputField.val());
```
do localStorage ukladejte nazvy clanku do pole v podobe JSONu pod vhodne nazvanym klicem napr. `localStorage.setItem('articleNames', '["Dragon", "Human"]')`

nvbach91 reviewed (changes requested):
Ukoly:
- pridat vyhledavani, (wikipedia opensearch api) a stahovat clanky na kliknuti na vysledky vyhledavani
- moznost ukladat tyto vysledky do localStorage pro rychly pristup
- krome stahovani umoznete nahled clanku v iframu ve stejne strance. hint: 
```js
$.ajax({
    url: 'https://en.wikipedia.org/api/rest_v1/page/pdf/Dragon/a4',
    xhrFields: { responseType: 'blob'},
}).done((resp) => {
    const blob = new Blob([resp], { type: 'application/pdf', title: 'Dragon' });
    const objectUrl = URL.createObjectURL(blob);
    window.open(objectUrl);
});

```

valm08 committed:
Add files via upload

Verze z obhajoby.

valm08 committed:
Add files via upload

Přidán iframe. Zobrazí se pouze po načtení 1. PDF souboru přes tlačítko „Preview PDF“.

nvbach91 reviewed (approved):
![how-to-create-presentation-slides-that-are-out-of-this-world-by-slidecomet-itseugenec-kaixinspeaking-28-638](https://user-images.githubusercontent.com/20724910/104008962-8b66ea80-51aa-11eb-9b47-d64347f9797a.jpg)
